### PR TITLE
Switch edx-analytics-dashboard from nose to pytest

### PR DESCRIPTION
**Pull request by andrey-canon**
Move nose tests to pytest. https://openedx.atlassian.net/projects/BOM/issues/BOM-1359

### Packages removed
nose-exclude==0.5.0	
nose-ignore-docstring==0.2	
nose==1.3.7
django-nose==1.4.4

### New packages
atomicwrites==1.3.0       # via pytest
attrs==19.3.0             # via pytest
more-itertools==5.0.0     # via pytest
packaging==20.1           # via pytest
pluggy==0.13.1            # via pytest
py==1.8.1                 # via pytest

### Reviewers
- [x] @morenol 
- [x]  Is this ready for edX's review?
- [ ] @jmbowman 

### Warnings

nose is added by needle but this was removed as an explicit requirement

### Post-review
- [ ] Rebase and squash commits

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @andrey-canon! I've created [OSPR-4130](https://openedx.atlassian.net/browse/OSPR-4130) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by natabene** [comment]
@andrey-canon Thank you for your contribution. Is this work in progress or is this ready for our review?

**Comment by andrey-canon** [comment]
@natabene  @jmbowman this is ready for your review

File: requirements/local.txt [hunk 1]
```diff
@@ -63,28 +64,34 @@ logilab-common==1.4.0     # via -r requirements/test.txt
 logutils==0.3.4.1         # via -r requirements/test.txt
 mccabe==0.6.1             # via -r requirements/test.txt, pylint
 mock==1.3.0               # via -r requirements/test.txt, sure
+more-itertools==5.0.0     # via -r requirements/test.txt, pytest
 needle==0.5.0             # via -r requirements/test.txt, bok-choy
 newrelic==5.8.0.136       # via -r requirements/test.txt, edx-django-utils
-nose-exclude==0.5.0       # via -r requirements/test.txt
-nose-ignore-docstring==0.2  # via -r requirements/test.txt
-nose==1.3.7               # via -r requirements/test.txt, django-nose, needle, nose-exclude
+nose==1.3.7               # via -r requirements/test.txt, needle
```
Comment (jmbowman) [inline requirements/local.txt:70]: Upgrading bok-choy should let you get rid of this and `needle`, but that can wait for a later PR.

**Comment by natabene** [comment]
@jmbowman Ready for you

**Review by jmbowman — APPROVED** [review]
Looks like it needs a rebase, but otherwise good to go.

**Comment by andrey-canon** [comment]
@jmbowman  rebased

**Comment by openedx-webhooks** [comment]
@andrey-canon 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-analytics-dashboard&entry.867055334=https://github.com/edx/edx-analytics-dashboard/pull/930&entry.1484655318=https://github.com/andrey-canon&entry.752974735=2020-02-28+22:42&entry.1917517419=2020-03-03+20:27&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**